File: src/database/connection.py
import configparser
import pyodbc

# Lee la configuración desde el archivo INI
config = configparser.ConfigParser()
config.read('./config/databaseConfig.ini')

db_config = config['database']

# Configuración de la conexión
connection_string = (
    f"DRIVER={{ODBC Driver 17 for SQL Server}};"
    f"SERVER={db_config['server']};"
    f"DATABASE={db_config['database']};"
    f"UID={db_config['user']};"
    f"PWD={db_config['password']};"
    f"PORT={db_config['port']}"
)

def connect_to_database():
    try:
        conn = pyodbc.connect(connection_string)
        print("Conexión exitosa a la base de datos")
        return conn
    except Exception as e:
        print(f"Error al conectar a la base de datos: {e}")
        raise

File: src/database/connection.js
const fs = require('fs');
const sql = require('mssql');

// Lee la configuración desde el archivo JSON
const configPath = './config/databaseConfig.json';
const dbConfig = JSON.parse(fs.readFileSync(configPath, 'utf8'));

// Configuración de la conexión
const sqlConfig = {
  user: dbConfig.user,
  password: dbConfig.password,
  server: dbConfig.server,
  database: dbConfig.database,
  port: dbConfig.port,
  options: {
    encrypt: true, // Usa true si estás usando Azure
    trustServerCertificate: true // Cambia según sea necesario
  }
};

// Función para conectarse a la base de datos
async function connectToDatabase() {
  try {
    const pool = await sql.connect(sqlConfig);
    console.log('Conexión exitosa a la base de datos');
    return pool;
  } catch (err) {
    console.error('Error al conectar a la base de datos:', err);
    throw err;
  }
}

module.exports = { connectToDatabase };
